src/ert/_c_wrappers/analysis/configuration.py

```diff
@@ -47,7 +47,7 @@ def check_parameters(cls, values):
         return values
 
     @validator("observations", each_item=True, pre=True)
-    def check_arguments(cls, observation):
+    def check_arguments(cls, observation: Union[str, List[str], Dict[str, Any]]) -> Any:
         """Because most of the time the users will configure observations as only a name
         we convert positional arguments to named arguments"""
         if isinstance(observation, str):
```

Comment:
Hard to see this in debugger, maybe the dict[str, Any] is the same typed as the others in here?

Reply:
ref
![Screenshot 2023-07-31 at 09 44 50](https://github.com/equinor/ert/assets/11595637/6758e1e4-da7b-4f48-8da7-775540485f4c)

could be 
```python
class ObservationDict(TypedDict):
    name: str
    index_list: NotRequired[List[int]]
``` 
Giving 

```python
def check_arguments(
        cls, observation: Union[str, List[str], ObservationDict]
    ) -> ObservationDict:
```